src/core/lib/security/transport/client_auth_filter.cc

```diff
@@ -165,17 +155,20 @@ static void on_credentials_metadata(void* arg, grpc_error_handle input_error) {
   grpc_auth_metadata_context_reset(&calld->auth_md_context);
   grpc_error_handle error = GRPC_ERROR_REF(input_error);
   if (error == GRPC_ERROR_NONE) {
-    GPR_ASSERT(calld->md_array.size <= MAX_CREDENTIALS_METADATA_COUNT);
+    GPR_ASSERT(calld->md_array->size() <= MAX_CREDENTIALS_METADATA_COUNT);
     GPR_ASSERT(batch->send_initial_metadata);
     grpc_metadata_batch* mdb =
         batch->payload->send_initial_metadata.send_initial_metadata;
-    for (size_t i = 0; i < calld->md_array.size; ++i) {
-      add_error(&error, grpc_metadata_batch_add_tail(
-                            mdb, &calld->md_links[i],
-                            GRPC_MDELEM_REF(calld->md_array.md[i])));
+    for (const auto& md : *calld->md_array) {
+      mdb->Append(md.first.as_string_view(), md.second.Ref(),
+                  [&](absl::string_view error, const grpc_core::Slice& value) {
```

Comment:
Prior to this PR, it looks like the code was returning an error in this case, not just logging.  Is this a behavior change?

Reply:
I think accidental... it's a different error that could be generated, but it should still generate.

Reply:
(fixed)